### ISSUE TITLE
Access authentication method

### ISSUE THREAD

**Issue by Mera-balou** [EtixLabs/RTSPAllTheThings#47]
Hi,

I 've just discovered your wonderful project and i have an enhancement to request.
It seems that when you set up a username and a password, like this example command :
`docker run --rm -e RTSP_USERNAME=admin -e RTSP_PASSWORD=1234 -p 8554:8554 ullaakut/rtspatt
`,
RTSPAll request by default a basic access authentication method.

In the RTSP RFC, both basic and digest access authentication methods are allowed.
Is it possible to add an override options to choose which one we want (default:basic if not set), like
 `[-e RTSP_AUTH=digest|basic]
`

Thanks

**Comment by Ullaakut** [comment]
Hi @Mera-balou !

I'm glad you find RTSTAPP useful :)

This sounds like quite a straight forward feature to implement, I'll get to it ASAP! 👍

**Comment by Ullaakut** [comment]
Alright so I've started an implementation and for this we'll need to build our own version of the GST RTSP server library, unfortunately, so this is going to take me some time, but I'm on it!

**Comment by Ullaakut** [comment]
Aaaand here you go: https://github.com/EtixLabs/RTSPAllTheThings/pull/48

I merged the PR, I will update the official docker image later this morning. I tested the feature on my machine and it seems to work fine 👌.

<img width="245" alt="screen shot 2018-06-07 at 08 02 26" src="https://user-images.githubusercontent.com/6976628/41080950-2789624c-6a29-11e8-86d1-60fa4708700b.png">

<img width="948" alt="screen shot 2018-06-07 at 08 02 48" src="https://user-images.githubusercontent.com/6976628/41080962-337f11c8-6a29-11e8-8394-22314c7e3065.png">

Let me know if that works for you as well, and if you have other feature requests, don't hesitate to tell me!

I know that there aren't a lot of options to serve RTSP and that the ones that exist are not very user friendly, so I really hope that with a bit more work on RTSPATT, we'll be able to help a significant amount of people. 😄
